docs/Manual/Deployment/Kubernetes/Drain.md

```diff
@@ -158,8 +158,8 @@ Check that each instance has a `Status` field with the value `"GOOD"`.
 Here is a shell command which makes this check easy, using the
 [`jq` JSON pretty printer](https://stedolan.github.io/jq/):
 
-```bash
-curl -k https://arangodb.9hoeffer.de:8529/_admin/cluster/health --user root: | jq . | grep '"Status"' | grep -v '"GOOD"'
+```
+curl -k https://localhost:8529/_admin/cluster/health --user root: | jq "[.Health | to_entries[] | select(.value.Status!=\"GOOD\") | {(.key): .value.Status }] | add"
```

Comment:
Demo input:

```json
{
  "Health": {
    "PRMR-2bd562cb-8f11-40fe-9c43-e318fa0fc65d": {
      "Endpoint": "tcp://[::1]:9929",
      "LastAckedTime": "2019-02-21T13:03:47Z",
      "SyncTime": "2019-02-21T13:03:47Z",
      "Version": "3.5.0-devel",
      "Engine": "rocksdb",
      "ShortName": "DBServer0001",
      "Timestamp": "2019-02-21T13:03:47Z",
      "Status": "GOOD",
      "SyncStatus": "SERVING",
      "Host": "e92663355763428d87d02231cea07c23",
      "Role": "DBServer",
      "CanBeDeleted": false
    },
    "PRMR-cdd978f8-e9ec-4fe8-be02-b53bd1e25144": {
      "Endpoint": "tcp://[::1]:9930",
      "LastAckedTime": "2019-02-21T13:03:47Z",
      "SyncTime": "2019-02-21T13:03:47Z",
      "Version": "3.5.0-devel",
      "Engine": "rocksdb",
      "ShortName": "DBServer0002",
      "Timestamp": "2019-02-21T13:03:47Z",
      "Status": "DEAD",
      "SyncStatus": "SERVING",
      "Host": "e92663355763428d87d02231cea07c23",
      "Role": "DBServer",
      "CanBeDeleted": false
    },
    "CRDN-3269c2a5-6b39-4dba-9bd4-d6ed7c36acab": {
      "Endpoint": "tcp://[::1]:9830",
      "LastAckedTime": "2019-02-21T13:03:45Z",
      "SyncTime": "2019-02-21T13:03:45Z",
      "Version": "3.5.0-devel",
      "Engine": "rocksdb",
      "ShortName": "Coordinator0001",
      "Timestamp": "2019-02-21T13:03:45Z",
      "Status": "GOOD",
      "SyncStatus": "SERVING",
      "Host": "e92663355763428d87d02231cea07c23",
      "Role": "Coordinator",
      "CanBeDeleted": false
    },
    "PRMR-ae1a9707-3a96-45d1-ae90-5f21c736929d": {
      "Endpoint": "tcp://[::1]:9931",
      "LastAckedTime": "2019-02-21T13:03:48Z",
      "SyncTime": "2019-02-21T13:03:48Z",
      "Version": "3.5.0-devel",
      "Engine": "rocksdb",
      "ShortName": "DBServer0003",
      "Timestamp": "2019-02-21T13:03:48Z",
      "Status": "GOOD",
      "SyncStatus": "SERVING",
      "Host": "e92663355763428d87d02231cea07c23",
      "Role": "DBServer",
      "CanBeDeleted": false
    },
    "AGNT-0328877f-63c0-4692-8aa9-c5dc7054d672": {
      "Endpoint": "tcp://[::1]:5301",
      "Role": "Agent",
      "CanBeDeleted": false,
      "Version": "3.5.0-devel",
      "Engine": "rocksdb",
      "Leader": "AGNT-0328877f-63c0-4692-8aa9-c5dc7054d672",
      "Leading": true,
      "LastAckedTime": 0,
      "Status": "TIMEOUT"
    }
  },
  "ClusterId": "dc11afde-c4bd-4701-82a1-8edd99136c63",
  "error": false,
  "code": 200
}
```

Output:
```json
{
  "PRMR-cdd978f8-e9ec-4fe8-be02-b53bd1e25144": "DEAD",
  "AGNT-0328877f-63c0-4692-8aa9-c5dc7054d672": "TIMEOUT"
}
```

Reply:
I have changed the wrong sentence about "the one pod.

The idea with jq is good, but I am unsure about a platform-independent way of quoting things, so I have not yet implemented this. 

The `kubectl` commands are `kubectl cordon <nodename>` and `kubectl uncordon <nodename>`, therefore, I think `uncordoned` is ok.

Cleaning out manually is optional, since the operator does a cleanout automatically, we might be able to improve on this later on, and we probably have to, since GKE will on node-upgrade just move on beyond our control. This is work in progress.

You are right about the difficulties with the grace period.